sessiongate/pkg/apis/sessiongate/v1alpha1/types.go

```diff
@@ -0,0 +1,220 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+type Session struct {
+	metav1.TypeMeta `json:",inline"`
+
+	// metadata is a standard object metadata
+	// +optional
+	metav1.ObjectMeta `json:"metadata,omitempty,omitzero"`
+
+	// spec defines the desired state of Session
+	// +required
+	Spec SessionSpec `json:"spec"`
+
+	// status defines the observed state of Session
+	// +optional
+	Status SessionStatus `json:"status,omitempty,omitzero"`
+}
+
+// +kubebuilder:validation:XValidation:rule="self == oldSelf",message="spec is immutable"
+type SessionSpec struct {
+	// ttl is the time-to-live duration for the session
+	// +kubebuilder:validation:Required
+	TTL metav1.Duration `json:"ttl"`
+
+	// managementCluster specifies the AKS management cluster
+	// +kubebuilder:validation:Required
+	ManagementCluster ManagementCluster `json:"managementCluster"`
+
+	// hostedControlPlane specifies the hosted control plane
+	// +kubebuilder:validation:Required
+	HostedControlPlane HostedControlPlane `json:"hostedControlPlane"`
+
+	// accessLevel defines the access permissions for the session
+	// +kubebuilder:validation:Required
+	AccessLevel AccessLevel `json:"accessLevel"`
+
+	// owner identifies the principal (user or service account) that owns this session
+	// +kubebuilder:validation:Required
+	Owner Principal `json:"owner"`
+}
+
+// ManagementCluster identifies an Azure management cluster
+type ManagementCluster struct {
+	// resourceId is the Azure resource ID of the management cluster
+	// +kubebuilder:validation:Required
```

Comment:
Look into using CEL for immutable constraints - do you really want to allow mutations to this field after creation? 

See example: https://github.com/openshift/hypershift/blob/351f6ae4538229f30733e474f1596fdc9813b073/api/certificates/v1alpha1/certificaterevocationrequest_types.go#L28